### PR TITLE
refactor(wizard): typed Response DTOs for SetupWizardController AJAX actions (slice 21 — REC #5b)

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Closes the audit gap REC #5 left open after the slice-13 `TaskController` split: `SetupWizardController` was the last backend controller still returning ad-hoc associative arrays from every `JsonResponse`. Migrates **all ten ad-hoc-array call sites** in `detectAction`, `testAction`, `discoverAction`, `generateAction`, and `saveAction` (five success paths + five error/exception paths) to typed Response DTOs, matching the precedent set by `ConfigurationController`, `ProviderController`, and the post-split task controllers.

## New Response DTOs (5)

| DTO | Used in | Wire shape |
|-----|---------|------------|
| `ProviderDetectionResponse` | `detectAction` | `{success, provider}` |
| `WizardTestConnectionResponse` | `testAction` | `{success, message}` |
| `DiscoveredModelsResponse` | `discoverAction` | `{success, models[]}` |
| `GeneratedConfigurationsResponse` | `generateAction` | `{success, configurations[]}` |
| `WizardSaveResponse` | `saveAction` | `{success, message, provider:{uid,name}, modelsCount, configurationsCount}` |

All errors continue to use the existing `Response/ErrorResponse`. Each DTO follows the established `final readonly` + `implements JsonSerializable` + typed `jsonSerialize()` pattern.

### Why a slim `WizardTestConnectionResponse` (not the existing `TestConnectionResponse`)

The existing `TestConnectionResponse` (used by `ConfigurationController::testProviderConnectionAction()`) carries an additional `models[]` field. The wizard's `testAction` only confirms reachability — model discovery is the dedicated `discoverAction` step — so a narrower DTO keeps the wire body byte-for-byte identical to the pre-DTO literal rather than introducing an unused `models: []` field.

## Wire-format preservation

The JSON shape consumed by `Backend/SetupWizard.js` (reads `result.success`, `result.provider`, `result.message`, `result.models`, `result.configurations`, `result.modelsCount`, `result.configurationsCount`, `result.provider.name`, `result.error`) is preserved byte-for-byte. Each new DTO's `jsonSerialize()` returns exactly the array shape the previous inline literal produced; key order matches the pre-DTO literal order.

## Test coverage (+13 tests, +44 assertions)

`Tests/Unit/Controller/Backend/Response/ResponseDtoTest` gains 13 tests covering the five new DTOs: shape assertions (key order via `array_keys()`), factory-method round-trips (`fromDetectedProvider`, `fromResult`, `fromDiscoveredModels`, `fromSuggestedConfigurations`, `fromProvider`), and edge cases (empty list factories, null-uid handling).

## Out of scope

REC #8b (replacing the surviving `catch (Throwable)` blocks in `saveAction`) is a separate slice and not touched here.

## Test plan

- [x] cgl
- [x] phpstan (level 10) — no errors
- [x] rector (dry-run) — clean
- [x] architecture (PHPat) — pass
- [x] unit — 3354 tests, 7298 assertions (3341 → 3354 = +13 tests, +44 assertions)
- [ ] CI matrix (PHP 8.2-8.5 × TYPO3 13.4 / 14.0)
- [ ] Smoke-test the wizard frontend manually (verify `Backend/SetupWizard.js` still consumes `result.provider`, `result.modelsCount`, etc. as expected)

### CODE REVIEW

File: Classes/Controller/Backend/Response/WizardSaveResponse.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    public static function fromProvider(
+        Provider $provider,
+        int $modelsCount,
+        int $configurationsCount,
+        string $message = 'Configuration saved successfully',
+    ): self {
+        return new self(
+            message: $message,
+            providerUid: $provider->getUid() ?? 0,
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/Backend/Response/WizardSaveResponse.php]: Fixed in commit 3aeb69f. Reverted the `?? 0` fallback — `providerUid` and `provider.uid` are nullable again, matching the pre-DTO wire shape and preserving the frontend's null-check semantics for unpersisted entities.

File: Tests/Unit/Controller/Backend/Response/ResponseDtoTest.php [hunk 2]
```diff
@@ -410,4 +424,202 @@ public function testErrorResponseKeyOrder(): void
...
+        // to (`?? 0`) when an unpersisted entity is passed.
+        $data = (new WizardSaveResponse(
+            message: 'Configuration saved successfully',
+            providerUid: 0,
+            providerName: 'Detached',
+            modelsCount: 1,
+            configurationsCount: 1,
+        ))->jsonSerialize();
+
+        self::assertSame(0, $data['provider']['uid']);
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Tests/Unit/Controller/Backend/Response/ResponseDtoTest.php]: Fixed in commit 3aeb69f. Replaced the misleading test with two real ones: `testWizardSaveResponseFromProviderPassesNullUidThrough` (instantiates `Provider` without uid → asserts `provider.uid: null`) and `testWizardSaveResponseFromProviderEmitsPersistedUid` (sets uid via `_setProperty` → asserts `42` round-trips).

File: Classes/Controller/Backend/SetupWizardController.php [hunk 3]
```diff
@@ -197,12 +200,11 @@ public function discoverAction(ServerRequestInterface $request): ResponseInterfa
             endpoint: $endpoint,
         );
 
-        $models = $this->modelDiscovery->discover($detected, $apiKey);
+        $models = array_values($this->modelDiscovery->discover($detected, $apiKey));
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/Backend/SetupWizardController.php]: Fixed in commit 3aeb69f. Dropped the redundant `array_values()` wrapper — the factory already reindexes.

File: Classes/Controller/Backend/SetupWizardController.php [hunk 4]
```diff
@@ -246,12 +248,13 @@ public function generateAction(ServerRequestInterface $request): ResponseInterfa
             );
         }
 
-        $configurations = $this->configurationGenerator->generate($detected, $apiKey, $models);
+        $configurations = array_values(
+            $this->configurationGenerator->generate($detected, $apiKey, $models),
+        );
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/Backend/SetupWizardController.php]: Fixed in commit 3aeb69f. Same — dropped the redundant wrapper from `generateAction`.

File: Classes/Controller/Backend/Response/DiscoveredModelsResponse.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ */
+final readonly class DiscoveredModelsResponse implements JsonSerializable
+{
+    /**
+     * @param list<array<string, mixed>> $models Models as produced by `DiscoveredModel::toArray()`
+     */
+    public function __construct(
+        public array $models,
+        public bool $success = true,
+    ) {}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/Backend/Response/DiscoveredModelsResponse.php]: Fixed in commit 3aeb69f. Constructor now asserts `array_is_list($models)` and throws `InvalidArgumentException` (code 1735300010) on associative input — the wire shape is locked to a JSON array.
